src/main/java/edu/harvard/iq/dataverse/authorization/providers/shib/ShibServiceBean.java

```diff
@@ -165,11 +176,115 @@ public BuiltinUser findBuiltInUserByAuthUserIdentifier(String authUserIdentifier
+            logger.warning(ex.toString());
+            return null;
+        }
+        if (mdqApiRequest == null) {
+            logger.warning("mdq api request was null");
+            return null;
+        }
+        try {
+            mdqApiRequest.connect();
+        } catch (IOException ex) {
+            logger.warning(ex.toString());
+            return null;
+        }
+        
+        XMLStreamReader xmlr = null;
+
+        try {
+            XMLInputFactory xmlFactory = javax.xml.stream.XMLInputFactory.newInstance();
+            xmlr =  xmlFactory.createXMLStreamReader(new InputStreamReader((InputStream) mdqApiRequest.getInputStream()));
+            
+            while ( xmlr.next() == XMLStreamConstants.COMMENT );
+            xmlr.require(XMLStreamConstants.START_ELEMENT, null, "EntityDescriptor");
+            
+            while (xmlr.hasNext()) {
+                int event = xmlr.next();
+                
+                if (event == XMLStreamConstants.START_ELEMENT) {
+                    String currentElement = xmlr.getLocalName();
+                    
+                    if ("".equals(currentElement)) {
+                        int eventType = xmlr.next();
+                        if (eventType == XMLStreamConstants.CHARACTERS) {
+                            String affiliation = xmlr.getText();
+                            return affiliation;
+                        } else {
+                            logger.warning("Unexpected contet in the OrganizationDisplayName element");
+                            return null; 
+                        }
+                    }
+                    
+                } else if (event == XMLStreamConstants.END_ELEMENT) {
+                    if (xmlr.getLocalName().equals("EntityDescriptor")) return null;
+                }       
+            }
+            
+        } catch (IOException ioex) {
+            logger.warning("IOException instantiating a stream reader of the mdq api output" + ioex.getMessage());
+        } catch (XMLStreamException xsex) {
+            logger.warning("Failed to parse the xml output of the mdq api; " + xsex.getMessage());
+        } finally {
+            if (xmlr != null) {
+                try {
+                    logger.fine("closing xml reader");
+                    xmlr.close();
+                } catch (XMLStreamException xsex) {
+                    // we don't care at this point 
+                }
+            }
+        }
+
+        logger.warning("Failed to find an affiliation for " + shibIdp);
+        return null;
+    }
+
+    /*
+     * This is the old-style method of obtaining the affiliation - by calling 
+     * DiscoFeed, provided either by the locally-running "real" shibd instance, 
+     * or a static list in the same json format when in dev./testing mode. 
+     * It is kept in the code for now, under the assumption that somebody 
+     * may still have reasons to keep using the DiscoFeed-based model.    
```

Comment:
```suggestion
     * It is kept in the code for now because the majority of Dataverse 
     * installations using Shib are likely not using InCommon.    
```

In the release notes we say that if you're not using InCommon you should set `dataverse.feature.shibboleth-use-discofeed=true`. I'm pretty sure Harvard Dataverse and UNC Dataverse are the only ones using InCommon. All the other shib installations (of which there are many, I think) will operate in this non-InCommon mode (DiscoFeed-mode), as far as I understand.